pcdsdevices/delay_generator.py

```diff
@@ -0,0 +1,141 @@
+"""
+Delay generator class
+
+This module contains classes related to the SRS DG645 delay generator.
+"""
+
+from time import sleep
+
+from ophyd import Device
+from ophyd import Component as Cpt
+from ophyd import EpicsSignalRO
+from ophyd import EpicsSignal
+
+from pcdsdevices.interface import BaseInterface
+
+
+CHANNELS = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'T0']
+TRIGGER_SOURCES = {
+    '0': 'Internal',
+    '1': 'Ext ^edge',
+    '2': 'Ext ~edge',
+    '3': 'SS ext ^edge',
+    '4': 'SS ext ~edge',
+    '5': 'Single Shot',
+    '6': 'Line'
+}
+TRIGGER_INHIBITS = {
+    '0': 'Off',
+    '1': 'Triggers',
+    '2': 'AB',
+    '3': 'AB,CD',
+    '4': 'AB,CD,EF',
+    '5': 'AB,CD,EF,GH'
+}
+
+
+class Dg_channel(BaseInterface, Device):
+    """
+    Delay generator single channel class
+
+    Parameters
+    ----------
+    prefix: str
+        Base PV for the delay generator channel
+
+    name: str
+        Alias of the channel
+    """
+    delay = Cpt(EpicsSignal, 'DelayAO', kind='hinted')
+    delay_rbk = Cpt(EpicsSignalRO, 'DelaySI', kind='normal')
```

Comment:
You can also combine PVs that form a read-write pair into a single component, e.g.:
`delay = Cpt(EpicsSignal, 'DelaySI', write_pv='DelayAO', kind=normal)`

Reply:
Need to see how the string PV ends being handled when scanning this component with the DAQ etc. I would not be surprised that such a string would break something down the line here.
(This mostly a note to myself for when I can take the xpp daq)

Reply:
Hmm yeah mixing two data types for the read/write will almost certainly have consequences